src/transport/protocols.ts

```diff
@@ -0,0 +1 @@
+export const enum Protocol { HTTPBinary, HTTPStructured, HTTP };
```

Comment:
For small files like these, it might be worth it to inline it rather than have a new file.

For example, we can inline the contents into the `index.ts` file.